server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -395,11 +425,35 @@ public void persist(final Committer committer)
   {
     final List<Pair<FireHydrant, Interval>> indexesToPersist = Lists.newArrayList();
     for (Sink sink : sinks.values()) {
+      FireHydrant hydrant = sink.swap();
+      if (hydrant != null) {
+        indexesToPersist.add(Pair.of(hydrant, sink.getInterval()));
+      }
+    }
+    persist(committer, indexesToPersist);
+  }
+
+  public void persistOldest(final Committer committer)
+  {
+    Sink oldestSink = null;
+    long oldestAccessTime = -1;
+    for (Sink sink : sinks.values()) {
       if (sink.swappable()) {
-        indexesToPersist.add(Pair.of(sink.swap(), sink.getInterval()));
+        long lastAccessTime = sink.getLastAccessTime();
+        if (oldestAccessTime < 0 || lastAccessTime < oldestAccessTime) {
+          oldestAccessTime = lastAccessTime;
+          oldestSink = sink;
+        }
       }
     }
+    FireHydrant hydrant = oldestSink == null ? null : oldestSink.swap();
+    if (hydrant != null) {
+      persist(committer, Arrays.asList(Pair.of(hydrant, oldestSink.getInterval())));
```

Comment:
This is defeating the purpose of the Committer - it's meant to be a way we can either fire a callback or store some metadata when everything up to the current row has been persisted to disk. Selectively persisting sinks would lead to committing kafka offsets, acking rabbitmq messages, etc, for data that is not actually on disk yet.

That being said, the handling of Committers in RealtimePlumber is a bit sketchy anyway (there's a couple other situations where committing doesn't happen at the right time) so I'm not sure how fussy we should be about this.